src/lib/helloworld/main_module.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# License stuff
+require 'yast'
+
+Yast.import 'UI'
+Yast.import 'Label'
+
+# Simple Hello World! fully functioning example
+# @note This module can be changed using the YaST development tool
+#   "create-new-package"
+module HelloWorld
+  # Simple class for the MainDialog
+  class MainDialog
+    include Yast::UIShortcuts
+    include Yast::I18n
+    include Yast::Logger
+    
+    # Display the dialog
+    def run
+      textdomain 'hello_world'
+      return unless create_dialog
+
+      begin
+        event_loop
+      ensure
+        Yast::UI.CloseDialog
+      end
+    end
+
+    # Simple event loop
+    def event_loop
+      loop do
+        input = Yast::UI.UserInput
+        next unless input == :hello
+
+        log.info "Received #{input}"
+        # Break the loop
+        break
+      end
+    end
+
+    # Draws the dialog
+    def create_dialog
+      # Simple ycp-ui dialog box
+      Yast::UI.OpenDialog(
+        Opt(:decorated, :defaultsize),
+        # ycp-ui VBox widget to hold more widgets
+        VBox(
+          # ycp-ui Heading widget
+          Heading(_('New Module')),
+          # ycp-ui Lable widget
+          Label(_('Hello, World!')),
+          # nested ycp-ui VBox widget
+          VBox(
```

Comment:
why this vbox is needed for single element?

Reply:
Added a label to show more complexity.